src/boot/efi/drivers.c

```diff
@@ -0,0 +1,150 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <efi.h>
+#include <efilib.h>
+
+#include "drivers.h"
+#include "util.h"
+
+static VOID efi_unload_image(EFI_HANDLE *h) {
+        if (*h)
+                (VOID) uefi_call_wrapper(BS->UnloadImage, 1, *h);
+}
+
+static EFI_STATUS load_one_driver(
+                EFI_HANDLE parent_image,
+                EFI_LOADED_IMAGE *loaded_image,
+                const CHAR16 *fname) {
+
+        _cleanup_(efi_unload_image) EFI_HANDLE image = NULL;
+        _cleanup_freepool_ EFI_DEVICE_PATH *path = NULL;
+        _cleanup_freepool_ CHAR16 *spath = NULL;
+        EFI_STATUS err;
+
+        assert(parent_image);
+        assert(loaded_image);
+        assert(fname);
+
+        spath = PoolPrint(L"\\EFI\\systemd\\drivers\\%s", fname);
+        if (!spath)
+                return log_oom();
+
+        path = FileDevicePath(
+                        loaded_image->DeviceHandle,
+                        spath);
```

Comment:
Any particular reason you're wrapping this aggressively in drivers.c?

Reply:
This specific case for FileDevicePath() was a bit too much I guess, will remove.

But in general, if functions have really a lot of args, i.e. are complicated "centerpiece" calls I think it's nicer to break them.